suite-common/wallet-core/src/fiat-rates/fiatRatesReducer.ts

```diff
@@ -64,23 +70,22 @@ const updateLastWeekRates = (state: CoinFiatRates[], payload: LastWeekRates) =>
     }
 };
 
-const fiatRatesReducer = (state: State = initialState, action: Action): State =>
-    produce(state, draft => {
-        switch (action.type) {
-            case STORAGE.LOAD:
-                draft.coins = action.payload.fiatRates;
-                break;
-            case RATE_REMOVE:
-                remove(draft.coins, action.payload);
-                break;
-            case RATE_UPDATE:
-                updateCurrentRates(draft.coins, action.ticker, action.payload);
-                break;
-            case LAST_WEEK_RATES_UPDATE:
-                updateLastWeekRates(draft.coins, action.payload);
-                break;
-            // no default
-        }
-    });
+export const prepareFiatRatesReducer = createReducerWithExtraDeps(
+    fiatRatesInitialState,
+    (builder, extra) => {
+        builder
+            .addCase(extra.actionTypes.storageLoad, extra.reducers.storageLoadFiatRates)
```

Comment:
nit: I would probably prefer to have all cases that use `extra` at the end, like we did it in other PRs